src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -34,11 +34,11 @@ public class InstanceManager : IInstanceManager
         private readonly HttpClient _client;
         private readonly IScriptWebHostEnvironment _webHostEnvironment;
 
-        public InstanceManager(IOptionsFactory<ScriptApplicationHostOptions> optionsFactory, HttpClient client, IScriptWebHostEnvironment webHostEnvironment,
+        public InstanceManager(IOptionsFactory<ScriptApplicationHostOptions> optionsFactory, IHttpClientFactory clientFactory, IScriptWebHostEnvironment webHostEnvironment,
             IEnvironment environment, ILogger<InstanceManager> logger, IMetricsLogger metricsLogger, IMeshServiceClient meshServiceClient, IRunFromPackageHandler runFromPackageHandler,
             IPackageDownloadHandler packageDownloadHandler)
         {
-            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _client = clientFactory.CreateClient() ?? throw new ArgumentNullException(nameof(clientFactory));
```

Comment:
If `clientFactory` is `null` here, you'll get an exception.

Use : `clientFactory?.CreateClient()` instead